### PR TITLE
EDSC-3937: Implement Vite

### PR DESCRIPTION
**Pull request by daniel-zamora**
# Overview

### What is the feature?

Vite replaces webpack for bulding and starting the server, esbuild replaces webpack for serverless.

### What is the Solution?

- Updated `package.json` and various configuration files.
- Renamed component `.js` files to `.jsx`.
- Prebundled portal JSON files into one during build to fix issues with dynamic import.
- Modified CSS files to directly import dependencies, and fixes formatting of some css.
- Removed aliases from webpack configurations.
- Combined HTML partials into a single `index.html`.

### What areas of the application does this impact?

Build, testing, deployment, application code, css, configurations.  Most everything is potentially impacted.

# Testing

### Reproduction steps

Follow normal build, start, deployment and testing processes as their interfaces shouldn't have changed

# Checklist

- [x] I have added automated tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

### CODE REVIEW

File: serverless.yml [hunk 1]
```diff
@@ -188,6 +191,21 @@ custom:
...
+  # Serverless esbuild configurations
+  esbuild:
+    loader:
+      .svg: text
+    external: ['knex', 'pg']
+    bundle: true
+    minify: ${self:custom.minifyBuild.${self:provider.stage}}
+
   # Serverless Webpack configurations
   webpack:
```
Comment (daniel-zamora) [inline serverless.yml:210]: Remove these configs

File: serverless-configs/aws-functions.yml [hunk 2]
```diff
@@ -24,8 +24,8 @@
   migrateDatabase:
     handler: serverless/src/migrateDatabase/handler.default
     timeout: 300
-    webpack:
```
Comment (daniel-zamora) [inline serverless-configs/aws-functions.yml:27]: Remove these

File: serverless/src/edlLogin/handler.js [hunk 3]
```diff
@@ -27,6 +27,7 @@ const edlLogin = async (event) => {
     redirectUriPath
   } = getEarthdataConfig(earthdataEnvironment)
 
+  console.log('NODE_ENV', process.env.NODE_ENV)
```
Comment (eudoroolivares2016) [inline serverless/src/edlLogin/handler.js:30]: Is this on purpose?

Comment (daniel-zamora) [inline serverless/src/edlLogin/handler.js:30]: remove

File: serverless.yml [hunk 4]
```diff
@@ -9,6 +9,7 @@ provider:
   endpointType: PRIVATE
   memorySize: 128
   environment:
+    NODE_ENV: ${env:NODE_ENV, 'development'}
```
Comment (daniel-zamora) [inline serverless.yml:12]: is this needed?

Comment (macrouch) [reply serverless.yml:12]: Yes, it is needed

File: static/src/js/actions/__tests__/authToken.test.js [hunk 5]
```diff
@@ -3,6 +3,11 @@ import thunk from 'redux-thunk'
 import nock from 'nock'
 import * as tinyCookie from 'tiny-cookie'
 
+jest.mock('tiny-cookie', () => ({
```
Comment (daniel-zamora) [inline static/src/js/actions/__tests__/authToken.test.js:6]: check if still needed

File: static/src/js/App.jsx [hunk 6]
```diff
@@ -260,3 +260,5 @@ class App extends Component {
 }
 
 export default App
+
```
Comment (daniel-zamora) [inline static/src/js/App.jsx:263]: remove extra lines

File: static/src/js/components/AccessMethod/EchoForm.scss [hunk 7]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (daniel-zamora) [inline static/src/js/components/AccessMethod/EchoForm.scss:1]: extra line

File: package.json [hunk 8]
```diff
@@ -18,58 +18,70 @@
...
     "stop:cache": "docker-compose down",
+    "jest:prepare-ci": "bin/cypress-prepare-ci",
+    "prestart-ci": "npm run build",
+    "start-ci": "npm run preview",
     "cypress:open": "cypress open",
     "cypress:run": "cypress run -c video=false",
-    "cypress:ci": "start-server-and-test start:ci http-get://localhost:8080 cypress:run",
+    "cypress:ci": "start-server-and-test start-ci http-get://localhost:8080 cypress:run",
     "cypress:prepare-ci": "bin/cypress-prepare-ci",
     "lint": "eslint .",
```
Comment (macrouch) [inline package.json:43]: "lint": "eslint . --ext .jsx,.js",

File: test-env.js [hunk 9]
```diff
@@ -4,6 +4,10 @@
 // jQuery
 import nock from 'nock'
 import enableHooks from 'jest-react-hooks-shallow'
+import { TextEncoder, TextDecoder } from 'util'
```
Comment (daniel-zamora) [inline test-env.js:7]: double check jest tests work without these

File: jest.config.cjs [hunk 10]
```diff
@@ -1,8 +1,8 @@
 // https://github.com/ai/nanoid/issues/363#issuecomment-1140906651
-const esModulesToIgnore = ['nanoid'].join('|')
+const esModulesToIgnore = ['nanoid', 'lodash-es', 'sinon'].join('|')
```
Comment (eudoroolivares2016) [inline jest.config.cjs:2]: Are these needed? We aren't using those packages (sinon and nanoid) in the package.json

File: tests/e2e/authentication/__mocks__/authenticated_collections.js [hunk 11]
```diff
@@ -0,0 +1,3701 @@
+export default {
```
Comment (daniel-zamora) [inline tests/e2e/authentication/__mocks__/authenticated_collections.js:1]: these can mbe removed

File: tests/e2e/authentication/__mocks__/getSubscriptions.graphql.body.json [hunk 12]
```diff
@@ -1,7 +1,7 @@
 {
-  "data": {
```
Comment (daniel-zamora) [inline tests/e2e/authentication/__mocks__/getSubscriptions.graphql.body.json:2]: fix identation

File: tests/e2e/authentication/__mocks__/graphql.headers.json [hunk 13]
```diff
@@ -1,3 +1,3 @@
 {
-  "content-type": "application/json"
-}
+    "content-type": "application/json"
```
Comment (daniel-zamora) [inline tests/e2e/authentication/__mocks__/graphql.headers.json:2]: fix indentation

File: tests/e2e/collection_list/collection_list.spec.js [hunk 14]
```diff
@@ -5,7 +5,7 @@ import singleCollection from './__mocks__/single_collection.json'
 test.describe('Collection List Behavior', () => {
   test.beforeEach(async ({ page }) => {
     await page.route('**/*.{png,jpg,jpeg}', (route) => route.abort())
-    await page.route(/collections/, async (route) => {
+    await page.route(/collections.json/, async (route) => {
```
Comment (daniel-zamora) [inline tests/e2e/collection_list/collection_list.spec.js:8]: does $ work here?

File: tests/e2e/timeline/timeline.spec.js [hunk 15]
```diff
@@ -1,15 +1,15 @@
...
-import timeline from './__mocks__/timeline.json'
-import granules from './__mocks__/granules.json'
-import providers from './__mocks__/providers.json'
-import accessMethods from './__mocks__/access_methods.json'
-import collectionFixture from './__mocks__/authenticated_collections.json'
-
 import { login } from '../../support/login'
 import { getAuthHeaders } from '../../support/getAuthHeaders'
 
+const collectionsGraphJson = require('./__mocks__/collections_graph.json')
```
Comment (daniel-zamora) [inline tests/e2e/timeline/timeline.spec.js:6]: use import here

File: tests/e2e/timeline/timeline.spec.js [hunk 16]
```diff
@@ -72,7 +72,7 @@ test.describe('Timeline spec', () => {
       })
     })
 
-    await page.goto('/projects?p=!C1443528505-LAADS&sb=-77.15071678161621%2C38.78817179999825%2C-76.89801406860352%2C38.99784152603538&lat=37.64643450971326&long=-77.407470703125&zoom=7qt=2020-01-06T04%3A15%3A27.310Z%2C2020-01-13T07%3A32%3A50.962Z&ff=Map%20Imagery&tl=1563377338!4!!')
+    await page.goto('/projects?p=!C1443528505-LAADS&sb=-77.15071%2C38.78817%2C-76.89801%2C38.99784&lat=37.64643&long=-77.40747&zoom=7qt=2020-01-06T04%3A15%3A27.310Z%2C2020-01-13T07%3A32%3A50.962Z&ff=Map%20Imagery&tl=1563377338!4!!')
```
Comment (daniel-zamora) [inline tests/e2e/timeline/timeline.spec.js:75]: can we figure out why this happend?

File: jest.config.cjs [hunk 17]
```diff
@@ -1,8 +1,8 @@
...
 
 module.exports = {
-  collectCoverage: true,
+  collectCoverage: false,
```
Comment (daniel-zamora) [inline jest.config.cjs:5]: true

File: cypress/e2e/paths/search/granules/collection_details/collection_details.cy.js [hunk 18]
```diff
@@ -473,7 +473,7 @@ describe('Path /search/granules/collection-details', () => {
       cy.visit('/search/granules/collection-details?p=C1996546500-GHRC_DAAC')
 
       cy.get('.collection-details-minimap').within(() => {
-        cy.get('.leaflet-interactive').should('have.attr', 'd', 'M0 180L360 180L360 0L0 0L0 180z')
+        cy.get('.leaflet-interactive').should('have.attr', 'd', 'M0 180L360 180L360 0L0 0z')
```
Comment (daniel-zamora) [inline cypress/e2e/paths/search/granules/collection_details/collection_details.cy.js:476]: double check this test

File: static/src/js/actions/__tests__/authToken.test.js (hunk 5 above)
Comment (daniel-zamora) [reply static/src/js/actions/__tests__/authToken.test.js:6]: these tests will fail if this is removed

File: test-env.js (hunk 9 above)
Comment (daniel-zamora) [reply test-env.js:7]: note needed

File: tests/e2e/collection_list/collection_list.spec.js (hunk 14 above)
Comment (daniel-zamora) [reply tests/e2e/collection_list/collection_list.spec.js:8]: no

File: static/src/js/actions/__tests__/authToken.test.js (hunk 5 above)
Comment (macrouch) [reply static/src/js/actions/__tests__/authToken.test.js:6]: I'm still confused by why this is needed, but you can move this block to test-env.js and you don't have to mock the library in all the different tests, just once

File: jest.config.cjs (hunk 10 above)
Comment (daniel-zamora) [reply jest.config.cjs:2]: yes

File: tests/e2e/authentication/__mocks__/getSubscriptions.graphql.body.js
Comment (macrouch) [inline tests/e2e/authentication/__mocks__/getSubscriptions.graphql.body.js:1]: remove file

File: tests/e2e/authentication/__mocks__/graphql.headers.js
Comment (macrouch) [inline tests/e2e/authentication/__mocks__/graphql.headers.js:1]: remove file

File: tests/e2e/collection_list/collection_list.spec.js [hunk 19]
```diff
@@ -2,10 +2,10 @@ import { test, expect } from 'playwright-test-coverage'
 
 import singleCollection from './__mocks__/single_collection.json'
 
-test.describe('Collection List Behavior', () => {
+test.describe.only('Collection List Behavior', () => {
```
Comment (macrouch) [inline tests/e2e/collection_list/collection_list.spec.js:5]: remove .only

File: cypress/e2e/paths/search/granules/collection_details/collection_details.cy.js (hunk 18 above)
Comment (daniel-zamora) [reply cypress/e2e/paths/search/granules/collection_details/collection_details.cy.js:476]: yea this test is not right

File: tests/e2e/timeline/timeline.spec.js (hunk 16 above)
Comment (daniel-zamora) [reply tests/e2e/timeline/timeline.spec.js:75]: I rolled react-datetime back to   `"react-datetime": "^2.16.3",` and the test still passed. not completely sure whats going on here but it doesn't appear to be harmful

Comment (macrouch) [reply tests/e2e/timeline/timeline.spec.js:75]: I don't think we need to rollback react-datetime, these numbers are spatial drawings, or map lat/long

Comment (macrouch) [reply tests/e2e/timeline/timeline.spec.js:75]: Oh wait, these values are in a page.goto, so it is just what we are putting into the URL bar to load the page. I really don't know why they changed

**Review by trevorlang — APPROVED** [review]
Looking good to me. Nice work!

**Comment by eudoroolivares2016** [comment]
Testing in SIT
Vite testing:
- [x] Lambdas worked
- [x] Bundlers worked
- [x] Lazy laoding worked
- [x] S3 deployment worked
- [x] Shapefile subsetting worked
- [x] CSS looks good
- [x] Harmony
- [x] Echo-orders
- [x] ESI
- [x] Granule subsetting
- [x] Direct download
- [x] EDD integration
- [x] Browse-images on thumbnails (collection, and granule) and the pop up granule modal viewer
- [x] Database communication
- [x] Logging in (EDL is down in SIT but, high envs worked)
- [x] Temporal filters
- [x] Spatial filtering with all types
- [x] Shapefile Searching
- [x] State machine seems to be working
- [x] GTM appears on the tag manager preview and in the network tab
- [x] tried most of the other routes contact-info, preferences, subscriptions, etc
- [x] collection details, granule details
- [x] Cron jobs seemed to have fired for tagging (though of course not fully working in SIT for gibs)
- [x] Map imagery and colormaps are showing up
 -[x] tried a mix of safari and firefox
 -[x] Harmony variable tree subsetting looks good

**Review by eudoroolivares2016 — APPROVED** [review]

**Review by macrouch — APPROVED** [review]

**Pull request merged**